File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/view/activity/AuthActivity.java
package technology.mainthread.apps.gatekeeper.view.activity;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;
import technology.mainthread.apps.gatekeeper.R;
import technology.mainthread.apps.gatekeeper.databinding.ActivityAuthBinding;
import technology.mainthread.apps.gatekeeper.viewModel.AuthActivityViewModel;

public class AuthActivity extends DaggerAppCompatActivity {

    @Inject
    AuthActivityViewModel viewModel;

    public static Intent getIntent(Context context) {
        return new Intent(context, AuthActivity.class);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityAuthBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_auth);
        binding.setViewModel(viewModel);
        viewModel.initialize(this, binding);
    }

    @Override
    protected void onDestroy() {
        viewModel.terminate();
        super.onDestroy();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        viewModel.handleActivityResult(requestCode, data);
    }
}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/view/activity/AuthActivity.kt
package technology.mainthread.apps.gatekeeper.view.activity

import android.content.Context
import android.content.Intent
import androidx.databinding.DataBindingUtil
import android.os.Bundle
import dagger.android.support.DaggerAppCompatActivity
import technology.mainthread.apps.gatekeeper.R
import technology.mainthread.apps.gatekeeper.databinding.ActivityAuthBinding
import technology.mainthread.apps.gatekeeper.viewModel.AuthActivityViewModel
import javax.inject.Inject

fun getAuthIntent(context: Context): Intent {
    return Intent(context, AuthActivity::class.java)
}

class AuthActivity : DaggerAppCompatActivity() {

    @Inject
    internal lateinit var viewModel: AuthActivityViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityAuthBinding = DataBindingUtil.setContentView(this, R.layout.activity_auth)
        binding.viewModel = viewModel
        viewModel.initialize(this, binding)
    }

    override fun onDestroy() {
        viewModel.terminate()
        super.onDestroy()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        viewModel.handleActivityResult(requestCode, data)
    }

}
